Visual Studio compilation warning (x64)

1>.\tinyxml2.cpp(1687) : warning C4244: 'initializing' : conversion from '__int64' to 'int', possible loss of data
on the line
int delta = p - start;  // skip initial whitespace, BOM, etc.

--- a/tinyxml2.cpp
+++ b/tinyxml2.cpp
@@ -1684,7 +1684,7 @@ XMLError XMLDocument::Parse( const char* p, size_t len )
         return _errorID;
     }
 
-	int delta = p - start;	// skip initial whitespace, BOM, etc.
+    ptrdiff_t delta = p - start;	// skip initial whitespace, BOM, etc.
     ParseDeep( _charBuffer+delta, 0 );
     return _errorID;
 }

